sycl/include/sycl/reduction.hpp

```diff
@@ -871,18 +878,21 @@ bool reduCGFuncForRangeFastAtomics(handler &CGH, KernelType KernelFunc,
       Reducer.template atomic_combine(&Out[0]);
     }
   });
-  return Reduction::is_usm || Redu.initializeToIdentity();
+
+  if (Reduction::is_usm || Redu.initializeToIdentity())
+    reduction::withAuxHandler(CGH, [&](handler &CopyHandler) {
+      reduSaveFinalResultToUserMem<KernelName>(CopyHandler, Redu);
+    });
 }
 
 namespace reduction {
 namespace main_krn {
 template <class KernelName, class NWorkGroupsFinished> struct RangeFastReduce;
 } // namespace main_krn
 } // namespace reduction
-template <typename KernelName, typename KernelType, int Dims,
-          typename PropertiesT, class Reduction>
-bool reduCGFuncForRangeFastReduce(handler &CGH, KernelType KernelFunc,
-                                  const range<Dims> &Range,
+template <typename KernelName, typename KernelType, typename PropertiesT,
+          class Reduction>
+void reduCGFuncForRangeFastReduce(handler &CGH, KernelType KernelFunc,
```

Comment:
To avoid confusion, could we use another name instead of `KernelFunc`? Now that we include the reduction-loop as part of this is might create confusion if we're not being explicit about it.

Reply:
Discussed offline - will leave as-is for now. The reasoning is that I'd really like to invoke those routines from the nd_range version where kernel would be just a kernel and not some wrapper. 

I have to do some perf tuning/applicability checks first though.